cpp/Makefile

```diff
@@ -63,12 +63,12 @@ ci-image: hdoc-build
 
 .PHONY: ci
 ci: ci-image
-	docker run -t --rm -v $(CURDIR):/mcap/cpp mcap_cpp_ci ./build.sh
+	docker run -t --rm -v $(CURDIR):/mcap/cpp -v $(HOME)/.conan/data:/root/.conan/data mcap_cpp_ci ./build.sh
```

Comment:
Why do we want the user's local install of conan to interact with the version contained inside Docker?

Reply:
To take advantage of caching. The "user" here is `ci`

Reply:
I guess this makes sense as a way to allow it to be cached – but would caching the Docker layers also help?

Reply:
We already cache the Docker layers, and the protobuf library (conan dependency) is rebuilt on virtually every PR which is why CI is so slow.